File: solutions/dataStructures/arrays/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Initialize Hash Map
        prev = {}

        # One Pass Solution
        for i, n in enumerate(nums):
            difference = target - n
            if difference in prev:
                return [prev[difference], i]
            prev[n] = i

        return []

File: solutions/grind75/week1/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> prev = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int complement = target - nums[i];

            if (prev.containsKey(complement)) {
                return new int[]{prev.get(complement), i};
            }

            prev.put(nums[i], i);
        }

        return new int[]{};
    }
}